### PR TITLE
Patch for "no polynomial" case for azimuth FM rate mitigation & handling the burst without burst polygon

### PR DESCRIPTION
**Pull request by seongsujeong**
This PR is an attempt to fix the rare case that the list pf polynomials (azimuth FM rate or doppler centroid) does not have any overlap with the sensing start / stop of the burst. In that case, the azimuth FM rate mismatch correction finds the nearest polynomials in azimuth time, but the indices for polynomials (`index_start` and `index_end`) has the same value. When that happens interpolator from `InterpolatedUnivariateSpline` cannot be generated.

When that happens, the `index_start` and `index_end` are adjusted so that an`InterpolatedUnivariateSpline` gets to have valid parameters to create the interpolator.
When using the generated interpolator, optional parameter `ext=3` was set so that the interpolator returns the values at the left / right boundary values when the input parameter for the interpolation of out of range. This is equivalent to nearest neighbor in this case because the sensing start / stop is "out of range" from the list of polynomials' azimuth time.

Another issue that this PR handles is the when the burst does not have corresponding burst polygon. Such cases have been discovered as documented in #118. When that is the case, s1-reader will fill `Sentinel1BurstSlc.border` member with empty polygon. Likewise `Sentinel1BurstSlc.center` will be populated by an empty point geometry.

### CODE REVIEW

**Comment by seongsujeong** [comment]
@vbrancat @LiangJYu I think the PR is ready to review. I will submit a PR to COMPASS To change the code accordingly.

File: src/s1reader/s1_burst_slc.py [hunk 1]
```diff
@@ -911,15 +911,21 @@ def az_fm_rate_mismatch_from_llh(self,
...
+            return isce3.core.LUT2d()
         interpolator_tau0_ka = InterpolatedUnivariateSpline(
                                         fm_rate_aztime_sec_vec,
                                         self.extended_coeffs.fm_rate_tau0_vec,
+                                        ext=3,
                                         k=1)
         tau0_ka_interp = interpolator_tau0_ka(vec_t)[..., np.newaxis]
 
+        if len(dc_aztime_sec_vec) <=1:
+            return isce3.core.LUT2d()
```
Comment (seongsujeong) [inline src/s1reader/s1_burst_slc.py:924]: add more description about why empty LUT2 is being returned
Also some description about why such treatment was necesary(i.e. historical data w/ IPF 002.36)

File: src/s1reader/s1_reader.py [hunk 2]
```diff
@@ -14,6 +14,7 @@
 import isce3
 import numpy as np
 import shapely
+from shapely.affinity import translate
```
Comment (seongsujeong) [inline src/s1reader/s1_reader.py:17]: get rid of this module

**Review by vbrancat — APPROVED** [review]
LGTM @seongsujeong. We reviewed this PR during our OPERA CSLC tag-up meeting. Please, add comments in the code on which data is affected by this fix. Thank you :)

File: src/s1reader/s1_burst_slc.py [hunk 3]
```diff
@@ -911,15 +911,21 @@ def az_fm_rate_mismatch_from_llh(self,
                              for datetime_vec in self.extended_coeffs.dc_aztime_vec]
 
         # calculate splined interpolation of the coeffs. and tau_0s
+        if len(fm_rate_aztime_sec_vec) <=1:
```
Comment (LiangJYu) [inline src/s1reader/s1_burst_slc.py:914]:
```suggestion
        if len(fm_rate_aztime_sec_vec) <= 1:
```
nit - formatting. Same below

**Review by LiangJYu — APPROVED** [review]
LGTM. Only a nit comment

**Pull request merged**